falcon/routing/compiled.py

```diff
@@ -368,8 +370,21 @@ def _validate_template_segment(self, segment, used_names):
                     msg = 'Cannot instantiate converter "{}"'.format(name)
                     raise ValueError(msg) from e
 
-    def _generate_ast(self, nodes, parent, return_values, patterns, level=0, fast_return=True):
+    def _generate_ast(
+        self,
+        nodes: list,
+        parent,
+        return_values: list,
+        patterns: list,
+        params_stack: list,
```

Comment:
Is `params_stack` really a `list`?

A `list` does not normally expose a `.copy()` method.
However, later in this PR, we do `params_stack.copy()`.

Reply:
list has `.copy()`: https://docs.python.org/3/tutorial/datastructures.html

also I pass it here
https://github.com/falconry/falcon/blob/06e0eb549d2158396f8ba30f903e03660727b092/falcon/routing/compiled.py#L580-L586

Reply:
:man_facepalming: Nvm.